python/dgl/distributed/graph_partition_book.py

```diff
@@ -664,14 +700,28 @@ def __init__(self, part_id, num_parts, node_map, edge_map, ntypes, etypes):
         self._num_partitions = num_parts
         self._ntypes = [None] * len(ntypes)
         self._etypes = [None] * len(etypes)
+        self._canonical_etypes = [None] * len(etypes)
+        self._etype2canonical = {}
         for ntype in ntypes:
             ntype_id = ntypes[ntype]
             self._ntypes[ntype_id] = ntype
         assert all(ntype is not None for ntype in self._ntypes), \
                 "The node types have invalid IDs."
         for etype in etypes:
             etype_id = etypes[etype]
+            c_etype = None
+            if isinstance(etype, tuple):
+                assert len(etype) == 3, \
+                    'Canonical etype should be in format of (str, str, str).'
+                c_etype = etype
+                etype = etype[1]
             self._etypes[etype_id] = etype
+            self._canonical_etypes[etype_id] = c_etype
+            if c_etype is not None:
+                if etype in self._etype2canonical:
+                    self._etype2canonical[etype] = tuple()
```

Comment:
i'm confused. what does it do? reset the mapping from etype to canonical etype

Reply:
this mainly for the logic in `def to_canonical_etype()`:
```
if len(ret) == 0:
            raise DGLError('Edge type "%s" is ambiguous. Please use canonical edge type '
                            'in the form of (srctype, etype, dsttype)' % etype)
```